### PR TITLE
Add python test for SPMD+Runtime Python API

### PR DESCRIPTION
**Pull request by JackCaoG**

### CODE REVIEW

File: test/spmd/test_xla_spmd_python_api_interaction.py [hunk 1]
```diff
@@ -54,6 +55,55 @@ def test_xla_replication_devices(self):
     self.assertEqual(xm.xla_real_devices([device]), [device_type + ':0'])
 
 
+class BasicRuntimeAPITest(test_xla_sharding_base.XlaShardingTest):
+
+  @classmethod
+  def setUpClass(cls):
+    os.environ["XLA_USE_SPMD"] = "1"
```
Comment (jonb377) [inline test/spmd/test_xla_spmd_python_api_interaction.py:62]: Do we need to cover the `XLA_USE_SPMD=0` case, or is that covered by the existing tests?

**Review by jonb377 — APPROVED** [review]
LGTM, thanks Jack

File: test/spmd/test_xla_spmd_python_api_interaction.py (hunk 1 above)
Comment (JackCaoG) [reply test/spmd/test_xla_spmd_python_api_interaction.py:62]: I assumme all Non-SPMD case is covered by original test.

File: test/spmd/test_xla_spmd_python_api_interaction.py [hunk 2]
```diff
@@ -54,6 +55,55 @@ def test_xla_replication_devices(self):
...
+    device_type = os.environ['PJRT_DEVICE']
+    if device_type == "TPU":
+      self.assertGreater(xr.global_runtime_device_count(), 4)
+    elif device_type == "CPU":
+      self.assertEqual(xr.global_runtime_device_count(), 1)
+
+  def test_addressable_runtime_device_count(self):
+    device_type = os.environ['PJRT_DEVICE']
+    if device_type == "TPU":
+      self.assertGreater(xr.addressable_runtime_device_count(), 4)
```
Comment (JackCaoG) [inline test/spmd/test_xla_spmd_python_api_interaction.py:102]: oh it should be greater or equal.. let me update this

**Review by will-cromar — APPROVED** [review]

**Comment by JackCaoG** [comment]
ehh I have no idea why CI is not running..

**Pull request merged**